dotCMS/src/main/java/com/dotcms/webdav/FsFileResource.java

```diff
@@ -0,0 +1,133 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package com.dotcms.webdav;
+
+import com.dotcms.repackage.com.bradmcevoy.common.ContentTypeUtils;
+import com.dotcms.repackage.com.bradmcevoy.http.Auth;
+import com.dotcms.repackage.com.bradmcevoy.http.CopyableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.DeletableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.GetableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.MoveableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.PropFindableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.PropPatchableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.Range;
+import com.dotcms.repackage.com.bradmcevoy.http.Request;
+import com.dotcms.repackage.com.bradmcevoy.http.entity.PartialEntity;
+import com.dotcms.repackage.com.bradmcevoy.http.exceptions.NotFoundException;
+import com.dotcms.repackage.com.bradmcevoy.http.webdav.PropPatchHandler.Fields;
+import com.dotcms.repackage.com.bradmcevoy.io.ReadingException;
+import com.dotcms.repackage.com.bradmcevoy.io.WritingException;
+import com.dotcms.webdav.FileContentService;
+
+
+import java.io.*;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ */
+public class FsFileResource extends FsResource implements CopyableResource, DeletableResource, GetableResource, MoveableResource, PropFindableResource, PropPatchableResource {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') "CopyableResource" is implemented by a super class; there is no need to implement it here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1939)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') "MoveableResource" is implemented by a super class; there is no need to implement it here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1939)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 175 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)